core/src/main/java/io/seata/core/rpc/ChannelManager.java

```diff
@@ -490,4 +490,24 @@ private static Channel tryOtherApp(ConcurrentMap<String, ConcurrentMap<String, C
         return chosenChannel;
 
     }
+
+    /**
+     * get rm channels
+     *
+     * @return
+     */
+    public static Map<String,Channel> getRmChannels() {
+        if (RM_CHANNELS.isEmpty()) {
+            return null;
+        }
+        Map<String,Channel> channels = new HashMap(RM_CHANNELS.size());
+        for (String resourceId : RM_CHANNELS.keySet()) {
+            Channel channel = tryOtherApp(RM_CHANNELS.get(resourceId), null);
```

Comment:
why tryOtherApp？ tryOtherApp just get from RM_CHANNELS, loop is also getting from  RM_CHANNELS  and your map can only save one channel for a resource

Reply:
the method i have modified，when the param applitioncation is null，will find one avtive channel from all application.